dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/TrackerIdentifier.java

```diff
@@ -28,39 +28,56 @@
  * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.hisp.dhis.attribute.AttributeValue;
 import org.hisp.dhis.common.IdentifiableObject;
+import org.hisp.dhis.util.ObjectUtils;
 
 /**
- * @author Morten Olav Hansen <mortenoh@gmail.com>
+ * @author Stian Sandvold
  */
-public enum TrackerIdentifier
+
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class TrackerIdentifier
 {
-    /**
-     * Preheat using UID identifiers.
-     */
-    UID,
+    public final static TrackerIdentifier UID = builder().idScheme( TrackerIdScheme.UID ).build();
+
+    public final static TrackerIdentifier CODE = builder().idScheme( TrackerIdScheme.CODE ).build();
 
-    /**
-     * Preheat using CODE identifiers.
-     */
-    CODE,
+    public final static TrackerIdentifier AUTO = builder().idScheme( TrackerIdScheme.AUTO ).build();
 
-    /**
-     * Find first non-null identifier in order: UID, CODE
-     */
-    AUTO;
+    @Builder.Default
+    private TrackerIdScheme idScheme = TrackerIdScheme.UID;
+
+    @Builder.Default
+    private String value = null;
```

Comment:
Null should already be default for strings

Reply:
Yeah, but compilation fails due to Lombok expecting a explicit value.

Reply:
Oh, ok.. even without @Builder.Default ? because you dont need it unless you want to assign a default value